connector/src/main/java/org/geysermc/connector/entity/Entity.java

```diff
@@ -166,14 +182,52 @@ public void moveRelative(GeyserSession session, double relX, double relY, double
     }
 
     public void moveRelative(GeyserSession session, double relX, double relY, double relZ, Vector3f rotation, boolean isOnGround) {
+        Position bedLocation = new Position((int) position.getFloorX(), (int) position.getFloorY(), (int) position.getFloorZ());
+        bedId = session.getConnector().getWorldManager().getBlockAt(session, bedLocation);
+        bedRotationZ = BlockTranslator.getJavaIdBlockMap().inverse().get(bedId);
         setRotation(rotation);
         setOnGround(isOnGround);
         this.position = Vector3f.from(position.getX() + relX, position.getY() + relY, position.getZ() + relZ);
 
         MoveEntityAbsolutePacket moveEntityPacket = new MoveEntityAbsolutePacket();
         moveEntityPacket.setRuntimeEntityId(geyserId);
-        moveEntityPacket.setPosition(position);
-        moveEntityPacket.setRotation(getBedrockRotation());
+        //Sets Villager position and rotation when sleeping
+        if(!metadata.getFlags().getFlag(EntityFlag.SLEEPING)){
```

Comment:
Please use whitespace around the brackets `if (!metadata.getFlags().getFlag(EntityFlag.SLEEPING)) {`